scalafix-interfaces/src/main/java/scalafix/internal/interfaces/ScalafixCoursier.java

```diff
@@ -0,0 +1,73 @@
+package scalafix.internal.interfaces;
+
+import coursierapi.Module;
+import coursierapi.*;
+import coursierapi.error.CoursierError;
+import scalafix.interfaces.ScalafixException;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ScalafixCoursier {
+
+    private static List<URL> fetch(
+            List<Repository> repositories,
+            List<Dependency> dependencies,
+            ResolutionParams resolutionParams
+    ) throws ScalafixException {
+        List<URL> jars = new ArrayList<>();
+        try {
+            List<File> files = Fetch.create()
+                    .withRepositories(repositories.stream().toArray(Repository[]::new))
+                    .withDependencies(dependencies.stream().toArray(Dependency[]::new))
+                    .withResolutionParams(resolutionParams)
+                    .fetch();
+            for (File file : files) {
+                URL url = file.toURI().toURL();
+                jars.add(url);
+            }
+        } catch (CoursierError | MalformedURLException e) {
+            throw new ScalafixException("Failed to fetch " + dependencies + "from " + repositories, e);
+        }
+        return jars;
+    }
+
+    public static List<URL> scalafixCliJars(
+            List<Repository> repositories,
+            String scalafixVersion,
+            String scalaVersion
+    ) throws ScalafixException {
+        Dependency scalafixCli = Dependency.parse(
```

Comment:
TIL Dependency.parse 🤩

I have implemented this logic too many times, nice to have it built in!

Reply:
Indeed! The hardest is/was to find a good name for the input ([`dep`](https://github.com/coursier/interface/blob/d55842de4f86ceb82cf332d0ccb416a47a6726dc/interface/src/main/java/coursierapi/Dependency.java#L44) is not very clear). I saw `coordinates` was used in other places of Coursier source code so I went for this... Any suggestion is welcome!